enhancements/installer/openstack-no-fip-install.md

```diff
@@ -0,0 +1,136 @@
+
+### User Stories
+
+#### Provider Networks
+
+As an enterprise OpenStack cluster administrator, I want to use the IPI installer but my cluster does not support Floating IPs. I need a way to install the OpenShift IPI installer on provider networks instead. I may also need to clear IP addresses I use on the provider network through an internal audit.
+
+##### Example Usage
+
+Using the proposed feature, the user would not set `externalNetwork` or `lbFloatingIP`. Then, they will set`machinesSubnet` to the ID of an existing subnet on a provider network that they want to install the cluster onto, and set the `networking.machineNetwork` to the CIDR of that subnet. They will also set custom ports IPs for the `apiVP` and `ingressVIP` to ensure the IP they take is available. Here is an example install config:
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+networking:
+  machineNetwork:
+  - cidr: 10.0.1.0/17
+platform:
+  openstack:
+    cloud: mycloud
+    computeFlavor: m1.s2.xlarge
+    machinesSubnet: fa806b2f-abcd-4bce-b9db-124bc64209bf
+    apiVIP: `10.0.1.19`
+    ingressVIP: `10.0.1.20`
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+#### No Public External Access
+
+As an administrator with an on-prem OpenStack cluster, I want to install OpenShift in a way that does not expose API and Ingress endpoints to an external network. Instead, I would prefer to attach my own loadbalancers to the cluster entrypoints post install.
+
+
+### Test Plan   
+
+Unit tests and validations for this feature will be added to the installer to make sure that correct usage is enforced and that this feature does not hinder the usage of other features. To ensure GA readiness, it will be vetted by the QE team as well to make sure that it works with the followin use cases: self-signed certs, customer-provided networks, baremetal workers, scale-out, upgrades, and a restricted-network install.
+
+### Graduation Criteria
+
+This enhancement will follow standard graduation criteria.
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+#### Tech Preview -> GA
+
+- Community Testing
+- Sufficient time for feedback
+- Upgrade testing from 4.3 clusters utilizing this enhancement to later releases
+- Downgrade and scale testing are not relevant to this enhancement
+- E2E testing is not necessary for this feature
+
+### Infrastructure Needs
+
+- OpenStack cluster with provider networks
+  - PSI
+
+### Drawbacks
+
+We have to be very careful about how we implement this. Making features that were previously required optional may confuse users.
+We also need to be certian that users will not be able to mistakenly install a cluster that was intended to have external connectivity without it.
```

Comment:
maybe add an option to installer to take a system deployed by this method and attach it to an external network with api an ingress fips?

Reply:
The user can easily expose the VIP ports on an external network as a day 2 operation, by manually creating FIPs and assigning them to the API and Ingress ports.